python/pyarrow/_compute.pyx

```diff
@@ -1027,6 +1266,15 @@ cdef class _ScalarAggregateOptions(FunctionOptions):
 
 
 class ScalarAggregateOptions(_ScalarAggregateOptions):
+    __doc__ = f"""
+    Options for scalar aggregations.
+
+    Parameters
+    ----------
+    {_skip_nulls_doc()}
```

Comment:
Are we sure there is a strong guarantee that the options of `skip_nulls` and `min_count` will always forever match with those of `ScalarAggregate`? I mean, now they match, but I'm concerned that in 6 months we will have forgotten that the docstrings influence multiple classes and might accidentally add options that don't exist in the classes that reuse the existing docstrings.

Reply:
I'm not sure what you mean? `skip_nulls` and `min_count` are individual options, not functions.

Reply:
You mean that they will stay the same for ScalarAggregateOptions vs ElementWiseAggregateOptions vs ... (the different places it is being used)? 

It seems quite unlikely to me that the general description will not be correct anymore for one of those functions, but we can also always then update it if that would change (also if we all inline them duplicated, we need to think about updating the docstring if behaviour in C++ would change ..)

Reply:
Yes, I mean that there might be a point where the various places where they are used are not aligned anymore.
I'm not concerned about the current state of things, the functions report the docstrings of individual options etc, I'm mostly concerned that in 1 year from now we will forget how we designed that to be and someone might think that it's reasonable for example to add one more option in `_skip_nulls_doc()` causing a wrong option to be added to one of the Option classes using that function.

I guess one possible solution would be to have something inspecting the class and confirming the arguments in `__init__` match with those documented.

Reply:
> someone might think that it's reasonable for example to add one more option in _skip_nulls_doc() causing a wrong option to be added to one of the Option classes using that function.

Hmm, I don't understand. Why would someone add an option there?

Reply:
> I guess one possible solution would be to have something inspecting the class and confirming the arguments in `__init__` match with those documented. 

We have numpydoc validation check that already should do something like that? (I don't remember if it is ran by default, or is ran for the compute module)

But, that would also not really solve your concern that the "meaning" of the keyword would change for one of the functions (checking that the signature keywords match with the documented keywords doesn't guarantee anything about whether the description content is correct or not)

Reply:
> Hmm, I don't understand. Why would someone add an option there?

I suppose that Alessandro means something like the following: assume that at some point we add an additional accepted value to `skip_nulls`, and document it in the `skip_nulls_doc()` function. But that new value might only be accepted by a subset of the compute kernels that have a `skip_nulls` argument, and thus we would get an incorrect docstring for the others. 

Now, since `skip_nulls` is a boolean keyword (True/False) currently, it seems not that likely there will be a new accepted value being added.  
And again, if that happens, I think we can handle it at that time, and I think we will remember that this `skip_nulls_doc()` is used in several places, and thus have to verify if our changes are correct for all places where this is being used.

Reply:
You are right, the numpydoc check would catch a set of issues. I'm wondering btw if it's actually running given that it didn't catch https://github.com/apache/arrow/blob/master/python/pyarrow/array.pxi#L714

Reply:
> I'm wondering btw if it's actually running given that it didn't catch https://github.com/apache/arrow/blob/master/python/pyarrow/array.pxi#L714

I think the check for proper whitespace around the colon is not yet enabled. For now, https://github.com/apache/arrow/pull/7732 only enabled check "PR01", which checks if all parameters are present in the docstring

Reply:
On further inspection, it seems that `archery numpydoc` doesn't check class methods, opened https://issues.apache.org/jira/browse/ARROW-15321